nvflare/fuel/f3/drivers/uds_driver.py

```diff
@@ -0,0 +1,127 @@
+from nvflare.fuel.f3.drivers.base_driver import BaseDriver
+from nvflare.fuel.f3.drivers.driver import ConnectorInfo, Driver
+from nvflare.fuel.f3.drivers.driver_params import DriverCap, DriverParams
+from nvflare.fuel.f3.drivers.socket_conn import ConnectionHandler, SocketConnection
+
+log = logging.getLogger(__name__)
+
+
+class SocketStreamServer(ThreadingUnixStreamServer):
+    def __init__(self, path: str, driver: "Driver", connector: ConnectorInfo):
+        self.path = path
+        self.driver = driver
+        self.connector = connector
+        self.ssl_context = None  # SSL is not supported
+        self.local_addr = path
+
+        UnixStreamServer.__init__(self, path, ConnectionHandler, False)
+
+        try:
+            self.server_bind()
+            self.server_activate()
+        except BaseException as ex:
+            log.error(f"Error binding to the path {path}: {ex}")
+            self.server_close()
+            raise ex
+
+
+class UdsDriver(BaseDriver):
+    """Transport driver for Unix Domain Socket"""
+
+    def __init__(self):
+        super().__init__()
+        self.server = None
+
+    @staticmethod
+    def supported_transports() -> List[str]:
+        return ["uds"]
+
+    @staticmethod
+    def capabilities() -> Dict[str, Any]:
+        return {DriverCap.HEARTBEAT.value: False, DriverCap.SUPPORT_SSL.value: False}
+
+    def listen(self, connector: ConnectorInfo):
+        self.connector = connector
+
+        socket_path = self.get_socket_path(connector.params)
+        try:
+            os.unlink(socket_path)
+        except OSError:
+            if os.path.exists(socket_path):
+                raise CommError(CommError.ERROR, f"Can't remove existing socket file: {socket_path}")
+
+        self.server = SocketStreamServer(socket_path, self, connector)
+        self.server.serve_forever()
+
+    def connect(self, connector: ConnectorInfo):
+        self.connector = connector
+
+        socket_path = self.get_socket_path(connector.params)
+
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        sock.connect(socket_path)
+
+        connection = SocketConnection(sock, connector)
+        self.add_connection(connection)
+        connection.read_loop()
+        self.close_connection(connection)
+
+    def shutdown(self):
+        self.close_all()
+        if self.server:
+            self.server.shutdown()
+
+    @staticmethod
+    def get_urls(scheme: str, resources: dict) -> (str, str):
+
+        socket_path = resources.get("socket")
+        if not socket_path:
+            prefix = resources.get("socket_prefix")
+            if not prefix:
+                prefix = "/tmp/nvflare"
+
+            for i in range(10):
+                n = random.randint(1000, 999999)
+                socket_path = f"{prefix}_{n}"
+                if not os.path.exists(socket_path):
+                    break
+                log.debug(f"Socket path {socket_path} exists, retrying another one...")
+
```

Comment:
what if after we try 10 times we still can't find an available socket?
we should raise an error in that situation?